### PR TITLE
Update and refactor memory allocation/permission rules

### PR DESCRIPTION
**Pull request by mr-tz**
per the comments in #798 / https://github.com/mandiant/capa-rules/pull/798/files#r1264358655

### CODE REVIEW

File: host-interaction/process/inject/allocate-rwx-memory.yml
Comment (mr-tz) [inline host-interaction/process/inject/allocate-rwx-memory.yml:1]: now named: `allocate or change RWX memory`

File: lib/allocate-rw-memory.yml
Comment (mr-tz) [inline lib/allocate-rw-memory.yml:1]: now called: `allocate or change RW memory`

File: lib/allocate-memory.yml [hunk 1]
```diff
@@ -9,18 +9,24 @@ rule:
...
       - api: kernel32.VirtualAllocEx
       - api: kernel32.VirtualAllocExNuma
-      - api: kernel32.VirtualProtect
-      - api: kernel32.VirtualProtectEx
       - api: NtAllocateVirtualMemory
       - api: ZwAllocateVirtualMemory
       - api: NtMapViewOfSection
       - api: ZwMapViewOfSection
-      - api: NtProtectVirtualMemory
-      - api: ZwProtectVirtualMemory
```
Comment (mr-tz) [inline lib/allocate-memory.yml:26]: moved the removed APIs to `change memory protection`

File: load-code/shellcode/execute-shellcode-via-copyfile2.yml [hunk 2]
```diff
@@ -11,21 +11,7 @@ rule:
...
-            # mov     edx, 31Fh       ; dwSize
-            # mov     r8d, 1000h      ; flAllocationType
-            # lea     r9d, [rcx+40h]  ; flProtect
-            # call    cs:VirtualAlloc
-            - match: allocate memory
-            - or:
-              - number: 0x40
-              - instruction:
-                - mnemonic: lea
-                - offset: 0x40
```
Comment (mr-tz) [inline load-code/shellcode/execute-shellcode-via-copyfile2.yml:28]: moved to `allocate RWX memory` / `allocate RW memory`

**Comment by mr-tz** [comment]
@jtothej feel free to review and provide feedback here, as well

File: host-interaction/process/inject/inject-thread.yml [hunk 3]
```diff
@@ -15,8 +15,8 @@ rule:
   features:
     - and:
       - or:
-        - match: allocate RWX memory
-        - match: allocate RW memory
+        - match: allocate or change RWX memory
+        - match: allocate or change RW memory
```
Comment (williballenthin) [inline host-interaction/process/inject/inject-thread.yml:19]: does this exist? i might have missed it.

File: lib/allocate-memory.yml (hunk 1 above)
Comment (williballenthin) [reply lib/allocate-memory.yml:26]: this is a nice cleanup

**Review by williballenthin — APPROVED** [review]
this is a beautiful update that really shows how we can abstract logic with capa. thanks @mr-tz !

**Pull request merged**